backend/git_real/tests/test_activity_log_creators.py

```diff
@@ -102,4 +104,37 @@ def test_pushing(self, has_permission):
         self.assertEqual(entry.effected_user, push.repository.user)
         self.assertEqual(entry.object_1, push)
         self.assertEqual(entry.object_2, push.repository)
-        self.assertEqual(entry.event_type.name, creators.GIT_PUSH)
\ No newline at end of file
+        self.assertEqual(entry.event_type.name, creators.GIT_PUSH)
+
+
+class log_pr_opened_Tests(APITestCase):
+    @mock.patch.object(IsWebhookSignatureOk, "has_permission")
+    def test_pr_opened(self, has_permission):
+        has_permission.return_value = True
+
+        self.assertEqual(PullRequest.objects.count(), 0)
+
+        body, headers = get_body_and_headers("pull_request_opened")
+        pull_request_data=body["pull_request"]
+        url = reverse(views.github_webhook)
+        repo = RepositoryFactory(full_name=body["repository"]["full_name"])
+
+        self.client.post(url, format="json", data=body, extra=headers)
+
+        pull_request = PullRequest.create_or_update_from_github_api_data(repo, pull_request_data)
+        self.assertEqual(PullRequest.objects.count(), 1)
+         
+        entry = LogEntry.objects.last()
+        project = RecruitProject.objects.filter(
+                repository=pull_request.repository,
+            ).order_by("pk").last()
+
+
+        self.assertEqual(LogEntry.objects.count(), 1)
+
+        self.assertEqual(entry.actor_user, pull_request.user)
+        self.assertEqual(entry.effected_user, pull_request.user)
+        self.assertEqual(entry.object_1, pull_request)
+        self.assertEqual(entry.object_2, project)
+        self.assertEqual(entry.event_type.name, creators.PR_OPENED)
+        self.assertEqual(pull_request.state, 'open')
```

Comment:
So I made a breakpoint() above here since I wanted to verify if there were any other projects associated with the repo (because it would be good to make sure that the correct project is associated with object 2 in the case of more than one project per repo) but I discovered something interesting when running line by line:

`entry.actor_user == pull_request.user == None` (there is no return value so doesn't seem like there is anything associated with user)

similarly for
`entry.object_2 == project == None`
`entry.effected_user == pull_request.user == None`

So concerned the tests just passing because all those values are just null instead of actually verifying that they are the correct log entry item...

Reply:
It looks like this conversation is not resolved.

How about asserting that the user is not None?  The actor user and effected user need to both be defined.